truffleruby/src/main/ruby/core/process_mirror.rb

```diff
@@ -390,6 +390,10 @@ def spawn(options, command, arguments)
           pid = Truffle::Process.spawn command, arguments, env_array, options
           # Check if the command exists *after* invoking posix_spawn so we have a pid
           unless resolve_in_path(command)
+            if pid < 0
+              $? = ::Process::Status.new(pid, 127, nil, nil)
+              Errno.handle
```

Comment:
This will not show the name of the non-existing command I think.
Could you make a if/else and reuse the exception below which includes the command name?